xlml/apis/task.py

```diff
@@ -100,22 +118,37 @@ def provision(self) -> Tuple[DAGNode, airflow.XComArg, airflow.XComArg]:
 
       queued_resource_op, queued_resource_name = tpu.create_queued_resource(
           tpu_name,
-          self.task_test_config.accelerator,
           self.task_gcp_config,
           ssh_keys,
           self.tpu_create_timeout,
-          self.task_test_config.num_slices,
-      )
-      queued_resource_op >> tpu.ssh_tpu.override(task_id="setup")(
-          queued_resource_name,
-          # TODO(wcromar): remove split
-          self.task_test_config.setup_script,
-          ssh_keys,
-          self.all_workers,
+          self.task_test_config,
       )
 
+      # if use_startup_script is False, we need to enable setup with ssh.
```

Comment:
Can we have a func called `provision_with_startup_script`? so we keep the original provision as it is? 

So, in your run_with_startup_script, the dependency will be provision_with_startup_script >> post_process> clean_up

Reply:
Done